### DISCUSSION TITLE
Question: interest/feasibility of features based off of Angular release candidates

### DISCUSSION THREAD

**Discussion by michael-small** [nartc/ngxtension-platform#334] [Q&A] [answered]
## Question

Is a proposed/considered utility based off of a release candidate of Angular worth making an issue so early?

## Why I ask

I have been experimenting with an API introduced in v18.0.0-next.3. I believe it has great potential and I think I have a decent use case for it worth spinning out into a utility. That is its own discussion, but before I make a formal issue I want to make sure I am not proposing this too early. 

Ideally I want to make a good issue that is explained, has examples, tests, etc. But the underlying API is so new. I can hold off on making an issue until it is formally included in v18. But if the project would be open to issues based on an release candidate API, I would be glad to get this idea out there early and iterate on it.

edit: my particular idea and it's RC API aside, I am curious about if RC APIs are worth considering in this project in general.

## My proposed utility, since I am explaining my thoughts anyways

I am very excited for an upcoming feature in Angular that was introduced in [v18.0.0-next.3](https://github.com/angular/angular/releases/tag/18.0.0-next.3). The feature? [feat(forms): Unified Control State Change Events #54579](https://github.com/angular/angular/pull/54579). It closes an issue filed in Angular 2.0.0-rc.5, and is one of the most requested issues. As a reactive forms person, this is the coolest thing I am looking forward to in the future short of an official reactive forms + signal API.

I have made various utilities using this, including something like `valueChanges` and `statusChanges` for forms, but also for pristine and touched events. Ways of filtering the unified events by value/status/dirty/touched. And composed from those and more events, a function like this

```ts
function allEventsUnified<T>(form: AbstractControl<T>): Observable<{
    value: T;
    status: FormControlStatus;
    touched: boolean;
    pristine: boolean;
}>
```

Naturally, I have `toSignal`'d versions as well.

In my head, even if a bolted in new form with signals API somehow makes it into Angular 18, I still think there is potential in some of the things I am experimenting with.

**Comment by michael-small** [comment]
Note: things did change since I wrote this: event types changed some names and submit/reset events were added. But things look solid for v18 coming soon.

**Reply by eneajaho** [reply, marked as answer]
Hi, 
Yeah, this would be great if we could create some helper utils to unify all in one place. Looks like more stuff are added.

**Reply by michael-small** [reply]
Great. And for people who may be reading here, I linked to this discussion in this issue that formalizes this proposal: https://github.com/ngxtension/ngxtension-platform/issues/376

**Reply by michael-small** [reply]
I formally marked the PR for review. It has tests, example app, example stackblitz. And if it is considered for the library then I can get to works on the doc.
